app/src/main/java/com/sunasterisk/englishflashcard/data/source/local/dao/DictionaryDAOlmpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.sunasterisk.englishflashcard.data.source.local.dao
+
+import android.annotation.SuppressLint
+import android.content.ContentValues
+import android.content.Context
+import android.database.Cursor
+import android.database.sqlite.SQLiteDatabase
+import com.sunasterisk.englishflashcard.data.model.Dictionary
+import com.sunasterisk.englishflashcard.database.SQLiteHelper
+import com.sunasterisk.englishflashcard.database.SQLiteTable
+
+class DictionaryDAOlmpl private constructor(context: Context) : DictionaryDAO {
+
+    private val writableDatabase: SQLiteDatabase =
+        SQLiteHelper.getInstance(context).writableDatabase
+    private val readableDatabase: SQLiteDatabase =
+        SQLiteHelper.getInstance(context).readableDatabase
+
+    @SuppressLint("Recycle")
+    override fun getAllDictionaries(): List<Dictionary> {
+        val listDictionary = mutableListOf<Dictionary>()
+        val query = "SELECT * FROM " + SQLiteTable.TABLE_DICTIONARY
+        val cursor: Cursor = readableDatabase.rawQuery(query, null)
+        cursor.moveToFirst()
+        while (!cursor.isAfterLast) {
+            val dictionary = Dictionary(
+                cursor.getInt(0),
+                cursor.getString(1),
+                cursor.getString(2),
+                cursor.getString(3),
+                cursor.getString(4),
+                cursor.getInt(5)
+            )
+            listDictionary.add(dictionary)
+            cursor.moveToNext()
+        }
+        return listDictionary
+    }
+
+    override fun addDictionary(dictionary: Dictionary): Boolean {
+        val contenValue = ContentValues()
+        contenValue.put(SQLiteTable.COL_WORD, dictionary.word)
+        contenValue.put(SQLiteTable.COL_WORD_TYPE, dictionary.wordType)
+        contenValue.put(SQLiteTable.COL_SPELLING, dictionary.spelling)
+        contenValue.put(SQLiteTable.COL_TRANSLATE, dictionary.translate)
+        contenValue.put(SQLiteTable.COL_ID_TOPICS, dictionary.topicsId)
```

Comment:
```suggestion
        val contenValue = dictionary.getContentValues()
```
Tương tự, thêm hàm getContentValues() trong Dictionary cho gọn